File: solutions/csharp/squeaky-clean/1/SqueakyClean.cs
using System.Text;
public static class Identifier
{
    public static string Clean(string identifier)
    {
        StringBuilder strBuilder = new StringBuilder();

        for (int i = 0; i < identifier.Length; i++)
        {
            if (identifier[i] == ' ')
            {
                strBuilder.Append('_');
            }
            else if (identifier[i] == '-' && i + 1 < identifier.Length){
                strBuilder.Append(Char.ToUpper(identifier[i+1]));
                i++;
            }
            else if (identifier[i] == '\0'){
                strBuilder.Append("CTRL");
            }
            else if (char.IsLetter(identifier[i]) && (identifier[i] < 'α' || identifier[i] > 'ω')){
                strBuilder.Append(identifier[i]);
            }
        }
        return strBuilder.ToString();
    }
}

File: solutions/java/squeaky-clean/1/src/main/java/SqueakyClean.java
class SqueakyClean {
    static String clean(String identifier) {
        StringBuilder builder = new StringBuilder(identifier);

        for (int i = 0;  i < builder.length(); i++){
            if (Character.isDigit(builder.charAt(i)) || Character.isAlphabetic(builder.charAt(i))){
                switch(builder.charAt(i)){
                    case '4':
                        builder.setCharAt(i, 'a');
                        break;
                    case '3':
                        builder.setCharAt(i, 'e');
                        break;
                    case '0':
                        builder.setCharAt(i, 'o');
                        break;
                    case '1':
                        builder.setCharAt(i, 'l');
                        break;
                    case '7':
                        builder.setCharAt(i, 't');
                        break;
                    default:
                        break;
            }
            } else {
                if (builder.charAt(i) == ' ') builder.setCharAt(i, '_');
                else if (builder.charAt(i) == '-'){
                    builder.setCharAt(i+1, Character.toUpperCase(builder.charAt(i+1)));
                    builder.deleteCharAt(i);
                    i -= 1;
                }
                else {
                    builder.deleteCharAt(i);
                    i -= 1;
                }
            }

        }

        return builder.toString();
    }
}
